src/lib/headline_filter_parser.unit.test.js

```diff
@@ -17,9 +17,12 @@ describe('Headline filter parser', () => {
     });
   });
 
-  describe('Parsing intermediary states while typing', () => {
-    test('it does not crash when beginning to type a tag', () => {
-      expect(() => parser.parse(':')).not.toThrowError();
+  describe('Parser fails with invalid syntax but this is OK', () => {
+    test('Parser fails when beginning to type a tag', () => {
+      expect(() => parser.parse(':')).toThrowError();
```

Comment:
Just to confirm: It's going to be expected behavior for the parser to throw errors on intermediary states?

That's OK, imo. But then we should document the `parse` method accordingly and state that it is expected to throw exceptions which should be ignored.